File: terremoto/TerremotoBoss.java
// TERREMOTO – A MONTANHA QUE GRITA
// Simulação de buff de defesa após derrota
// Autor: Marcelo

import java.util.ArrayList;
import java.util.HashMap;

class Jogador {
    String nome;
    ArrayList<HashMap<String, String>> buffs = new ArrayList<>();

    public Jogador(String nome) {
        this.nome = nome;
    }
}

public class Terremoto {
    int buffDuration = 360; // 6 minutos

    public String derrotadoPor(Jogador j) {
        HashMap<String, String> buff = new HashMap<>();
        buff.put("nome", "Proteção de Terremoto");
        buff.put("vida_total", "+15%");
        buff.put("armadura", "+25%");
        buff.put("tempo_restante", buffDuration + "s");

        j.buffs.add(buff);
        return j.nome + " recebeu o buff de Terremoto por 6 minutos.";
    }

    public static void main(String[] args) {
        Jogador j = new Jogador("Herald3");
        Terremoto boss = new Terremoto();
        System.out.println(boss.derrotadoPor(j));
        System.out.println("Buffs ativos: " + j.buffs);
    }
}

File: terremoto/terremoto_boss.py
"""
TERREMOTO – A MONTANHA QUE GRITA
Simulação de buff defensivo após derrota
Autor: Marcelo
"""

import time

class Terremoto:
    def __init__(self):
        self.buff_duration = 360  # 6 minutos
        self.buff_active = False

    def derrotado_por(self, jogador):
        jogador["buffs"].append({
            "nome": "Proteção de Terremoto",
            "vida_total": "+15%",
            "armadura": "+25%",
            "tempo_restante": self.buff_duration
        })
        self.buff_active = True
        return f"{jogador['nome']} recebeu o buff de Terremoto por 6 minutos."

# Simulação de uso
if __name__ == "__main__":
    jogador = {"nome": "Herald3", "buffs": []}
    boss = Terremoto()
    print(boss.derrotado_por(jogador))
    print("Buffs ativos:", jogador["buffs"])
